browser/ui/webui/settings/brave_settings_leo_assistant_handler.cc

```diff
@@ -0,0 +1,141 @@
+  const auto item_visible_iter = FindAiChatSidebarItem(visible_items);
+
+  if (item_visible_iter != visible_items.end()) {
+    sidebar_service->RemoveItemAt(item_visible_iter - visible_items.begin());
+    return true;
+  }
+
+  return false;
+}
+
+}  // namespace
+
+namespace settings {
+
+BraveLeoAssistantHandler::BraveLeoAssistantHandler() = default;
+BraveLeoAssistantHandler::~BraveLeoAssistantHandler() = default;
+
+void BraveLeoAssistantHandler::RegisterMessages() {
+  profile_ = Profile::FromWebUI(web_ui());
+
+  web_ui()->RegisterMessageCallback(
+      "toggleLeoIcon",
+      base::BindRepeating(&BraveLeoAssistantHandler::HandleToggleLeoIcon,
+                          base::Unretained(this)));                          
+  web_ui()->RegisterMessageCallback(
+      "getLeoIconVisibility",
+      base::BindRepeating(&BraveLeoAssistantHandler::HandleGetLeoIconVisibility,
+                          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "resetLeoData",
+      base::BindRepeating(&BraveLeoAssistantHandler::HandleResetLeoData,
+                          base::Unretained(this)));
+}
+
+void BraveLeoAssistantHandler::OnJavascriptAllowed() {
+  sidebar_service_observer_.Reset();
+  sidebar_service_observer_.Observe(
+      sidebar::SidebarServiceFactory::GetForProfile(profile_));
+}
+
+void BraveLeoAssistantHandler::OnJavascriptDisallowed() {
+  sidebar_service_observer_.Reset();
+}
+
+void BraveLeoAssistantHandler::OnItemAdded(const sidebar::SidebarItem& item,
+                                           size_t index) {
+  if (item.built_in_item_type ==
+      sidebar::SidebarItem::BuiltInItemType::kChatUI) {
+    NotifyChatUiChanged(true);
+  }
+}
+
+void BraveLeoAssistantHandler::OnItemRemoved(const sidebar::SidebarItem& item,
+                                             size_t index) {
+  if (item.built_in_item_type ==
+      sidebar::SidebarItem::BuiltInItemType::kChatUI) {
+    NotifyChatUiChanged(false);
+  }
+}
+
+void BraveLeoAssistantHandler::NotifyChatUiChanged(const bool& isLeoVisible) {
+  if (!IsJavascriptAllowed()) {
+    return;
+  }
+  FireWebUIListener("settings-brave-leo-assistant-changed", isLeoVisible);
+}
+
+void BraveLeoAssistantHandler::HandleToggleLeoIcon(
+  const base::Value::List& args) {
+  auto* service = sidebar::SidebarServiceFactory::GetForProfile(profile_);  
+
+  if(!ShowLeoAssistantIconVisibleIfNot(service)) {
+    HideLeoAssistantIconIfNot(service);
+  }
+}
+
+void BraveLeoAssistantHandler::HandleGetLeoIconVisibility(
+  const base::Value::List& args) {
+  auto* service = sidebar::SidebarServiceFactory::GetForProfile(profile_);
+  const auto hidden_items = service->GetHiddenDefaultSidebarItems();
+  AllowJavascript();
+  ResolveJavascriptCallback(args[0],
+    !base::ranges::any_of(hidden_items, [](const auto& item) {
+        return item.built_in_item_type ==
+               sidebar::SidebarItem::BuiltInItemType::kChatUI;
+      }));
```

Comment:
tooootally optional: using base::Contains with projection might be more semantic.

```
!base::Contains(hidden_items,
                               sidebar::SidebarItem::BuiltInItemType::kChatUI,
                               &sidebar::SidebarItem::built_in_item_type));
```

Reply:
Changed `base::ranges::any_of` -> `base::Contains`